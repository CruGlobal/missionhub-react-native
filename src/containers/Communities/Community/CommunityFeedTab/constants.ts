export const COMMUNITY_FEED = 'nav/COMMUNITY_FEED';
