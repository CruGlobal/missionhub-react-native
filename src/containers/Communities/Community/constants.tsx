import React from 'react';
import i18next from 'i18next';

import GroupChallenges, {
  COMMUNITY_CHALLENGES,
} from '../../Groups/GroupChallenges';

import {
  CommunityFeedTab,
  COMMUNITY_FEED,
} from './CommunityFeedTab/CommunityFeedTab';
import {
  COMMUNITY_IMPACT,
  CommunityImpactTab,
} from './CommunityImpactTab/CommunityImpactTab';
import { CommunitiesCollapsibleHeaderContext } from './CommunityHeader/CommunityHeader';

// Moved here to prevent circular dependency
export const COMMUNITY_TABS = 'nav/COMMUNITY_TABS';

export const communityTabs = [
  {
    name: i18next.t('communityTabs:feed'),
    navigationAction: COMMUNITY_FEED,
    component: () => (
      <CommunityFeedTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
  {
    name: i18next.t('communityTabs:challenges'),
    navigationAction: COMMUNITY_CHALLENGES,
    component: GroupChallenges,
  },
  {
    name: i18next.t('communityTabs:impact'),
    navigationAction: COMMUNITY_IMPACT,
    component: () => (
      <CommunityImpactTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
];
