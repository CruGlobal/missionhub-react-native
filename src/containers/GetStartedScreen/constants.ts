export const GET_STARTED_SCREEN = 'nav/GET_STARTED';
