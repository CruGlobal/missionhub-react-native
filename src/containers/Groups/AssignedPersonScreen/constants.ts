export const CONTACT_PERSON_SCREEN = 'nav/CONTACT_PERSON';
export const IS_USER_CREATED_MEMBER_PERSON_SCREEN =
  'nav/IS_USER_CREATED_MEMBER_PERSON';
export const IS_GROUPS_MEMBER_PERSON_SCREEN = 'nav/IS_GROUPS_MEMBER_PERSON';
export const MEMBER_PERSON_SCREEN = 'nav/MEMBER_PERSON';
export const ME_PERSONAL_PERSON_SCREEN = 'nav/ME_PERSONAL_PERSON';
export const IS_GROUPS_ME_COMMUNITY_PERSON_SCREEN =
  'nav/IS_GROUPS_ME_COMMUNITY_PERSON';
export const ME_COMMUNITY_PERSON_SCREEN = 'nav/ME_COMMUNITY_PERSON';
