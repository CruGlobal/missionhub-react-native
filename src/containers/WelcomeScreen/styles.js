import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.primaryColor,
    paddingHorizontal: 60,
  },
  descriptionText: {
    color: theme.white,
    textAlign: 'left',
    paddingVertical: 10,
    fontSize: 24,
    lineHeight: 32,
  },
  headerText: {
    color: theme.secondaryColor,
    fontSize: 48,
  },
});
