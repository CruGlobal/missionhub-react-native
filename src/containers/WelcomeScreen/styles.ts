import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.primaryColor,
  },
  content: {
    paddingHorizontal: 60,
  },
  descriptionText: {
    color: theme.white,
    textAlign: 'left',
    paddingVertical: 10,
    fontSize: 24,
    lineHeight: 32,
  },
  headerText: {
    color: theme.secondaryColor,
    fontSize: 48,
  },
  filledButton: {
    backgroundColor: theme.secondaryColor,
    margin: 8,
    height: 48,
    width: theme.fullWidth - 80,
  },
  clearButton: {
    backgroundColor: theme.transparent,
    borderWidth: 1,
    borderColor: theme.secondaryColor,
    margin: 8,
    width: theme.fullWidth - 80,
    height: 48,
  },
  buttonText: {
    color: theme.white,
    fontWeight: '500',
    fontSize: 14,
    paddingTop: 2,
    letterSpacing: 1.5,
  },
});
