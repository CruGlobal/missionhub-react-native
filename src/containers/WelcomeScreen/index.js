import React, { Component } from 'react';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import PropTypes from 'prop-types';

import { navigatePush } from '../../actions/navigation';
import theme from '../../theme';
import { Flex, Text, Button } from '../../components/common';
import { SETUP_SCREEN } from '../SetupScreen';
import { disableBack } from '../../utils/common';
import { trackActionWithoutData } from '../../actions/analytics';
import { ACTIONS } from '../../constants';

import styles from './styles';

@translate('welcome')
class WelcomeScreen extends Component {
  componentDidMount() {
    disableBack.add();

    this.props.dispatch(trackActionWithoutData(ACTIONS.ONBOARDING_STARTED));
  }

  componentWillUnmount() {
    disableBack.remove();
  }

  navigateToNext = () => {
    const { dispatch, next } = this.props;

    // Remove the back handler when moving forward
    disableBack.remove();

    dispatch(next());
  };

  render() {
    const { t } = this.props;

    return (
      <Flex align="center" justify="center" value={1} style={styles.container}>
        <Flex value={4} align="start" justify="center">
          <Text type="header" style={styles.headerText}>
            {t('welcome')}
          </Text>
          <Text style={styles.descriptionText}>{t('welcomeDescription')}</Text>
        </Flex>

        <Flex value={1} align="stretch" justify="end">
          <Button
            type="secondary"
            onPress={this.navigateToNext}
            text={t('ok').toUpperCase()}
            style={{ width: theme.fullWidth }}
          />
        </Flex>
      </Flex>
    );
  }
}

WelcomeScreen.propTypes = {
  next: PropTypes.func.isRequired,
};

export default connect()(WelcomeScreen);
export const WELCOME_SCREEN = 'nav/WELCOME';
