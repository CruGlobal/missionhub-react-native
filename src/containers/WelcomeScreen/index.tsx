import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';
import { useTranslation } from 'react-i18next';
import { useNavigationParam } from 'react-navigation-hooks';
import { ThunkDispatch, ThunkAction } from 'redux-thunk';

import { Flex, Text, Button } from '../../components/common';
import BottomButton from '../../components/BottomButton';
import { trackActionWithoutData } from '../../actions/analytics';
import { ACTIONS } from '../../constants';
import Header from '../../components/Header';
import BackButton from '../BackButton';
import TrackOnFocus from '../TrackOnFocus';

import styles from './styles';

const WelcomeScreen = ({
  dispatch,
  next,
}: {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  dispatch: ThunkDispatch<any, null, never>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  next: (params: { signin: boolean }) => ThunkAction<void, any, null, never>;
}) => {
  useEffect(() => {
    dispatch(trackActionWithoutData(ACTIONS.ONBOARDING_STARTED));
  }, [dispatch]);

  const navigateToNext = (signin = false) => {
    dispatch(next({ signin }));
  };

  const signIn = () => {
    navigateToNext(true);
  };

  const allowSignIn = useNavigationParam('allowSignIn');
  const { t } = useTranslation('welcome');

  return (
    <View style={styles.container}>
      <TrackOnFocus screenNameFragments={['onboarding', 'welcome']} />
      <Header left={<BackButton />} />
      <Flex align="center" justify="center" value={1} style={styles.content}>
        <Flex value={3} align="start" justify="center">
          <Text header={true} style={styles.headerText}>
            {t('welcome')}
          </Text>
          <Text style={styles.descriptionText}>{t('welcomeDescription')}</Text>
        </Flex>

        {allowSignIn ? (
          <Flex value={1} align="center" justify="start">
            <Button
              testID="sign-in"
              pill={true}
              onPress={signIn}
              style={styles.filledButton}
              buttonTextStyle={styles.buttonText}
              text={t('signIn').toUpperCase()}
            />
            <Button
              testID="get-started-sign-in-variant"
              pill={true}
              onPress={navigateToNext}
              style={styles.clearButton}
              buttonTextStyle={styles.buttonText}
              text={t('getStartedButton').toUpperCase()}
            />
          </Flex>
        ) : (
          <BottomButton
            testID={'get-started'}
            onPress={navigateToNext}
            text={t('continue')}
          />
        )}
      </Flex>
    </View>
  );
};

export default connect()(WelcomeScreen);
export const WELCOME_SCREEN = 'nav/WELCOME';
