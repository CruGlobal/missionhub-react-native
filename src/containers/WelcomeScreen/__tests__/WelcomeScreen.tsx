import 'react-native';
import { fireEvent } from 'react-native-testing-library';

import WelcomeScreen from '..';

import { renderWithContext } from '../../../../testUtils';
import * as common from '../../../utils/common';
import { trackActionWithoutData } from '../../../actions/analytics';
import { ACTIONS } from '../../../constants';

const next = jest.fn(() => () => {});

jest.mock('react-native-device-info');
jest.mock('../../../actions/analytics');
jest.mock('../../../components/BottomButton', () => 'BottomButton');
jest.mock('../../../components/common', () => ({
  Flex: 'Flex',
  Text: 'Text',
  Button: 'Button',
}));

// @ts-ignore
common.disableBack = { add: jest.fn(), remove: jest.fn() };

(trackActionWithoutData as jest.Mock).mockReturnValue({
  type: 'tracked action without data',
});

describe('WelcomeScreen', () => {
  const baseConfig = { componentProps: { next } };
  const allowSignInVariantConfig = {
    ...baseConfig,
    navParams: { allowSignIn: true },
  };

  it('should render correctly', () => {
    renderWithContext(WelcomeScreen, baseConfig).snapshot();
  });

  it('should render correctly with sign in button', () => {
    renderWithContext(WelcomeScreen, allowSignInVariantConfig).snapshot();
  });

  it('getStarted btn should call next', () => {
    const { getByTestId } = renderWithContext(WelcomeScreen, baseConfig);
    fireEvent.press(getByTestId('get-started'));

    expect(common.disableBack.remove).toHaveBeenCalledTimes(1);
    expect(next).toHaveBeenCalledTimes(1);
    expect(next).toHaveBeenCalledWith({ signin: false });
  });

  it('tryItNow btn should call next', () => {
    const { getByTestId } = renderWithContext(
      WelcomeScreen,
      allowSignInVariantConfig,
    );
    fireEvent.press(getByTestId('get-started-sign-in-variant'));

    expect(common.disableBack.remove).toHaveBeenCalledTimes(1);
    expect(next).toHaveBeenCalledTimes(1);
    expect(next).toHaveBeenCalledWith({ signin: false });
  });

  it('signIn btn should call next with signIn = true', () => {
    const { getByTestId } = renderWithContext(
      WelcomeScreen,
      allowSignInVariantConfig,
    );
    fireEvent.press(getByTestId('sign-in'));

    expect(next).toHaveBeenCalledTimes(1);
    expect(next).toHaveBeenCalledWith({ signin: true });
  });

  it('should fire analytics event on mount', () => {
    renderWithContext(WelcomeScreen, baseConfig);

    expect(trackActionWithoutData).toHaveBeenCalledWith(
      ACTIONS.ONBOARDING_STARTED,
    );
  });

  it('should clean up back handler on unmount', () => {
    renderWithContext(WelcomeScreen, baseConfig).unmount();

    expect(common.disableBack.remove).toHaveBeenCalledTimes(1);
  });
});
