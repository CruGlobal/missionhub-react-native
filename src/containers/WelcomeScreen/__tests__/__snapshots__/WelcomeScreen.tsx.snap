// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <Flex
    align="center"
    justify="center"
    style={
      Object {
        "paddingHorizontal": 60,
      }
    }
    value={1}
  >
    <Flex
      align="start"
      justify="center"
      value={3}
    >
      <Text
        header={true}
        style={
          Object {
            "color": "#52C5DC",
            "fontSize": 48,
          }
        }
      >
        welcome!
      </Text>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 24,
            "lineHeight": 32,
            "paddingVertical": 10,
            "textAlign": "left",
          }
        }
      >
        Growing closer to God involves helping others experience Him. MissionHub joins you in that journey by suggesting steps of faith to take with others.
      </Text>
    </Flex>
    <BottomButton
      onPress={[Function]}
      testID="get-started"
      text="Let's Get Started"
    />
  </Flex>
</RCTSafeAreaView>
`;

exports[`WelcomeScreen should render correctly with sign in button 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#007398",
      "flex": 1,
    }
  }
>
  <Flex
    align="center"
    justify="center"
    style={
      Object {
        "paddingHorizontal": 60,
      }
    }
    value={1}
  >
    <Flex
      align="start"
      justify="center"
      value={3}
    >
      <Text
        header={true}
        style={
          Object {
            "color": "#52C5DC",
            "fontSize": 48,
          }
        }
      >
        welcome!
      </Text>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 24,
            "lineHeight": 32,
            "paddingVertical": 10,
            "textAlign": "left",
          }
        }
      >
        Growing closer to God involves helping others experience Him. MissionHub joins you in that journey by suggesting steps of faith to take with others.
      </Text>
    </Flex>
    <Flex
      align="center"
      justify="start"
      value={1}
    >
      <Button
        buttonTextStyle={
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 1.5,
            "paddingTop": 2,
          }
        }
        onPress={[Function]}
        pill={true}
        style={
          Object {
            "backgroundColor": "#52C5DC",
            "height": 48,
            "margin": 8,
            "width": 670,
          }
        }
        testID="sign-in"
        text="SIGN IN"
      />
      <Button
        buttonTextStyle={
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 1.5,
            "paddingTop": 2,
          }
        }
        onPress={[Function]}
        pill={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#52C5DC",
            "borderWidth": 1,
            "height": 48,
            "margin": 8,
            "width": 670,
          }
        }
        testID="get-started-sign-in-variant"
        text="GET STARTED"
      />
    </Flex>
  </Flex>
</RCTSafeAreaView>
`;
